lib/phubme/nicknames_matcher.ex

```diff
@@ -7,18 +7,19 @@ defmodule PhubMe.NicknamesMatcher do
     {:no_nicknames_found}
   end
 
-  defp matching_nicknames([ [ nickname ] | tail]) do
-    case nickname_from_mix_config(nickname) do
-      {:ok, matching_nickname} ->
-        [ [ nickname, matching_nickname ] | matching_nicknames(tail) ]
-      :error ->
-        IO.puts "No matching nickname found for " <> nickname
-        matching_nicknames(tail)
-    end
+  defp matching_nicknames(list, acc \\ [])
```

Comment:
This allows the call to the recursive method to be made as if it wasn't recursive at all. This is nice to keep the caller unaware of the implementation. Often, this is done by wrapping the recursive-function into another function with a clean interface.